packages/libncnn/build.sh

```diff
@@ -2,33 +2,118 @@ TERMUX_PKG_HOMEPAGE=https://github.com/Tencent/ncnn
 TERMUX_PKG_DESCRIPTION="A high-performance neural network inference framework optimized for the mobile platform"
 TERMUX_PKG_LICENSE="BSD 3-Clause"
 TERMUX_PKG_MAINTAINER="@termux"
-TERMUX_PKG_VERSION="20230223"
-TERMUX_PKG_REVISION=1
+_COMMIT=4b97730b0d033b4dc2a790e5c35745e0dbf51569
+TERMUX_PKG_VERSION="20230627"
 TERMUX_PKG_SRCURL=git+https://github.com/Tencent/ncnn
 TERMUX_PKG_GIT_BRANCH=master
+TERMUX_PKG_DEPENDS="abseil-cpp, glslang, libc++"
+TERMUX_PKG_BUILD_DEPENDS="protobuf-static, python, vulkan-headers, vulkan-loader-android"
+TERMUX_PKG_PYTHON_COMMON_DEPS="wheel, pybind11"
 TERMUX_PKG_BUILD_IN_SRC=true
-TERMUX_PKG_NO_STATICSPLIT=true
 TERMUX_PKG_EXTRA_CONFIGURE_ARGS="
--DCMAKE_BUILD_TYPE=Release
--DNCNN_VULKAN=ON
--DVulkan_LIBRARY=${TERMUX_PREFIX}/lib/libvulkan.so
--DVulkan_INCLUDE_DIRS=${TERMUX_PREFIX}/include
--DNCNN_PLATFORM_API=OFF
+-DNCNN_BUILD_BENCHMARK=OFF
 -DNCNN_BUILD_EXAMPLES=OFF
+-DNCNN_BUILD_TESTS=OFF
+-DNCNN_BUILD_TOOLS=OFF
+-DNCNN_DISABLE_EXCEPTION=OFF
 -DNCNN_DISABLE_RTTI=OFF
--DNCNN_BUILD_BENCHMARK=OFF
--DNCNN_SIMPLEOCV=ON
--DNCNN_SYSTEM_GLSLANG=ON
+-DNCNN_ENABLE_LTO=ON
 -DNCNN_OPENMP=ON
 -DNCNN_PYTHON=ON
--DCMAKE_EXE_LINKER_FLAGS=-ldl
+-DNCNN_SHARED_LIB=OFF
+-DNCNN_SIMPLEOCV=ON
+-DNCNN_SIMPLEOMP=ON
+-DNCNN_SIMPLESTL=OFF
+-DNCNN_SYSTEM_GLSLANG=ON
+-DNCNN_VULKAN=ON
+-DVulkan_LIBRARY=${TERMUX_PREFIX}/lib/libvulkan.so
+-DVulkan_INCLUDE_DIRS=${TERMUX_PREFIX}/include
 "
 
-TERMUX_PKG_BUILD_DEPENDS="glslang"
-TERMUX_PKG_DEPENDS="libc++, libprotobuf, vulkan-headers, vulkan-tools"
-TERMUX_PKG_PYTHON_COMMON_DEPS="wheel, pybind11"
+termux_step_post_get_source() {
+	git fetch --unshallow
+	git checkout "${_COMMIT}"
+	git submodule update --init --recursive
+}
```

Comment:
I doubt if this works as intended for submodules. Please use the method taken in `c-toxcore` or `telegram-bot-api` package. Also I recommend checking the SHA256 sum of the source tree.